tensorflow/compiler/mlir/tosa/transforms/legalize_tfl.cc

```diff
@@ -223,6 +224,61 @@ LogicalResult ConvertTFLReluOp::matchAndRewrite(
   return success();
 }
 
+LogicalResult ConvertTFLRelu1Op::matchAndRewrite(
+    Operation* op, PatternRewriter& rewriter) const {
+  auto tfl_relu1_op = cast<TFL::Relu1Op>(op);
+
+  ShapedType input_type = tfl_relu1_op.x().getType().dyn_cast<ShapedType>();
+  ShapedType output_type =
+      tfl_relu1_op.getResult().getType().dyn_cast<ShapedType>();
+  // Not a ranked tensor output
+  if (!input_type || !output_type) return failure();
+
+  bool input_is_qtype =
+      input_type.getElementType().isa<mlir::quant::UniformQuantizedType>();
+  bool output_is_qtype =
+      output_type.getElementType().isa<mlir::quant::UniformQuantizedType>();
+
+  if (input_is_qtype != output_is_qtype) {
+    return op->emitOpError(
+        "ConvertTFLRelu1Op: input/output tensor should "
```

Comment:
Note: if this invalid for the op, it should be flagged in verifier for op, if this is limitation of pattern then rewriter.notifyMatchFailure should be used. For error messages follow https://llvm.org/docs/CodingStandards.html#error-and-warning-messages wrt style.

Reply:
Yes we weighed how to go about with the use of notifyMatchFailure. It seems that would be better off as a separate overhaul of this whole file towards just that cleanup. This particular PR just replicates the relu6 code as a starter task. Do you prefer we fix here or as a separate task covering all the ops ?

Reply:
> Yes we weighed how to go about with the use of notifyMatchFailure. It seems that would be better off as a separate overhaul of this whole file towards just that cleanup. This particular PR just replicates the relu6 code as a starter task. Do you prefer we fix here or as a separate task covering all the ops ?

@jpienaar Hi Jacques, any thoughts and what's your opinion on this one? I personally suggest it would be better to do a separate PR for everything else together instead of doing this ad-hoc one? Tks! 🙂

Reply:
SG, sorry slipped through. Follow up to handle all SGTM.

Reply:
Great! Tks Jacques for reviewing my 1st PR for Tensorflow :)